lib/awslib/sts.go

```diff
@@ -0,0 +1,87 @@
+package awslib
+
+import (
+	"context"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/service/sts"
+)
+
+const expirationBuffer = 10 * time.Minute
+
+type Credentials struct {
+	// Generated:
+	awsAccessKeyID     string
+	awsSecretAccessKey string
+	awsSessionToken    string
+	expiresAt          time.Time
+
+	// Arguments to generate the credentials:
+	_awsAccessKeyID     string
+	_awsSecretAccessKey string
+	_awsRoleARN         string
+	_sessionLabel       string
+}
+
+func GenerateSTSCredentials(ctx context.Context, awsAccessKeyID string, awsSecretAccessKey string, roleARN string, sessionLabel string) (Credentials, error) {
```

Comment:
Added this so that we can reuse this in other places. The idea is that we'll need to use STS to generate temporary credentials if we're assuming a particular role.